### DISCUSSION TITLE
Should we consistently convert OSPF areas into IPv4 addresses

### DISCUSSION THREAD

**Discussion by ipspace** [ipspace/netlab#1194] [Features]
Right now, netlab checks the validity of OSPF areas (they could be IP addresses or 32-bit integers) but does not convert them into a unified format.

Would it make sense to convert integers into IP addresses (i.e. 0 into 0.0.0.0 or 1 into 0.0.0.1)? I vaguely remember someone already had to do that in a device configuration template.

What do you think @ssasso @jbemmel?

**Comment by ssasso** [comment]
You remember correctly, it has been done here:

https://github.com/ipspace/netlab/blob/48149d2b84bd4be908afaee1eb8eb5de119a50e3/netsim/ansible/templates/ospf/srlinux.macro.j2#L1

and also here:

https://github.com/ipspace/netlab/blob/48149d2b84bd4be908afaee1eb8eb5de119a50e3/netsim/ansible/templates/ospf/routeros7.ospf-include.j2#L3

I think we need to make sure other devices do not want the "reverse" to work ;)

**Comment by ipspace** [comment]
OK, will add the conversion into a test branch and run the multi-area OSPFv2 and OSPFv3 integration tests for all devices.

**Comment by ipspace** [comment]
Implemented in 938d3668. Tested on all device images I have available, all passed the ospfv2/02-areas integration test.
